bin/hdfs-mesos-datanode

```diff
@@ -2,17 +2,16 @@
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-JAVA_HOME_DIR=$(dirname $(readlink -f $(which java)))
-
-if [ -f $JAVA_HOME_DIR/../../bin/java ];
-then
-  export JAVA_HOME=$JAVA_HOME_DIR/../..
-elif [ -f $JAVA_HOME_DIR/../bin/java ];
-then
-  export JAVA_HOME=$JAVA_HOME_DIR/..
-else 
-  echo “Error JAVA_HOME cannot be determined from the registry”
-  exit 1;
+if [ -z "$JAVA_HOME" ]; then
+  JAVA_HOME_DIR=$(dirname $(readlink -f $(which java)))
+  if [ -f $JAVA_HOME_DIR/../../bin/java ]; then
+    export JAVA_HOME=$JAVA_HOME_DIR/../..
+  elif [ -f $JAVA_HOME_DIR/../bin/java ]; then
+    export JAVA_HOME=$JAVA_HOME_DIR/..
+  else 
+    echo “Error JAVA_HOME cannot be determined from the registry”
```

Comment:
I still don't know what the "registry" refers to. How about "Error: Could not determine JAVA_HOME"